uaclient/cli.py

```diff
@@ -133,10 +132,10 @@ def new_f(args, cfg, **kwargs):
             cfg.write_cache("lock", "{}:{}".format(os.getpid(), lock_holder))
             notice_msg = "Operation in progress: {}".format(lock_holder)
             cfg.add_notice("", notice_msg)
-            _LOCK_FILE = lock_file  # Set _LOCK_FILE for cleanup
+            _CLEAR_LOCK_FILE = cfg.delete_cache_key
             retval = f(args, cfg, **kwargs)
-            util.remove_file(lock_file)
-            cfg.remove_notice("", notice_msg)
+            cfg.delete_cache_key("lock")
```

Comment:
why a text string vs a constant or something here? Easy for typos across uses.

Reply:
should this be around a try and only _CLEAR_LOCK_FILE when we know it succeeds?

Reply:
> why a text string vs a constant or something here? Easy for typos across uses.

This is a good point, and I've taken a small tech-debt issue here so we can address this in 26.1 because there are sprawling changes throughout the code that really require that we create a strict defintion of allowed cache_keys
for all of the [UAConfig data_paths](https://github.com/canonical/ubuntu-advantage-client/blob/master/uaclient/config.py#L60-L68) . That'll be a broad and cosmetic change that I don't want to pollute this PR for 26.0 if that sounds ok?

Reply:
> should this be around a try and only _CLEAR_LOCK_FILE when we know it succeeds?

So, if either the retval = f(args, cfg, **kwargs) or  cfg.delete_cache_key("lock") fails with an unexpected error we immediately raise out of this context and don't get to the line where we set _CLEAR_LOCK_FILE = None. 
Once in [main_error_handler we have exception handling](https://github.com/canonical/ubuntu-advantage-client/pull/1339/files#diff-d558bdbcb7ee1bef85fd138367f1e64708ea8271c09cf8e4c19941c24c8883d6R942-R944) that looks to see if a dirty _CLEAR_LOCK_FILE callable is set. Only if _CLEAR_LOCK_FILE is still "dirty" and we didn't get there due to an existing LockHeldError, do we call _CLEAR_LOCK_FILE("lock") which removes the lock file as well as any notices related to that lock "Operation in progress.*" message.